### PR TITLE
Completed fetch-in-react-practice.

### PR DESCRIPTION
**Pull request by stefancarrera**
react-practice

### CODE REVIEW

File: fetch-in-react-practice/client/components/app.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    item.isCompleted = !item.isCompleted;
+    fetch(`api/todos/${todoId}`, {
+      method: 'PATCH',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(item)
+    })
+      .then(response => response.json())
+      .then(item => {
+        const updatedTodos = this.state.todos;
+        updatedTodos[item.id] = item;
```
Comment (scbowler) [inline fetch-in-react-practice/client/components/app.jsx:65]: This mutates the state, you need to make a new array before you can make changes to it. You can do the following to make a new array:

```JAVASCRIPT
const updatedTodos = this.state.todos.slice();
updatedTodos[item.id] = item;
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good! But one update needed

**Comment by stefancarrera** [comment]
Corrected the original state mutation from the toggleCompleted function

**Review by scbowler — APPROVED** [review]
Looks good! 👍

**Pull request merged**